### ISSUE TITLE
Why ID in Json Representation of Archetype?

### ISSUE THREAD

**Issue by azsam** [imulus/Archetype#300]
Hi all,

I know that's probably not the most appropriate place for the question - but I'd like to know:
Archetype's data is saved in a table cmsPropertyData in the field dataNtext as json. In the json representation there's also an attribute "id". Why is that needed? Can I generate it myself when migrating data from csv/json/whatever to Archetype?
Thanks!

**Comment by kgiszewski** [comment]
I believe @kjac added this for compatibility with `Upload` and `ImageCropper` datatypes. https://github.com/imulus/Archetype/commit/d4166f33e8038de4a3c0430eed901125c5221bc3

However if you need to generate an Archetype in code, you can probably ignore it and use a pattern like so: 

https://github.com/kgiszewski/ArchetypeManual/blob/master/06%20-%20Editing%20an%20Archetype%20in%20Code.md

I'll let @kjac chime in if there are any perils I'm unaware of ;)

**Comment by kjac** [comment]
Hey... You beat me to the punch, Kevin!

Off the top of my head I'd say the id's are generated automatically if
needed/absent. I can't get to the code right now, but if need be I can
check it one of the next days. But I think you should simply just go ahead
and try it out :-)

**Comment by azsam** [comment]
So it's safe to simply generate a bunch of GUID's and assign them as id's of the migrated json data?

**Comment by kgiszewski** [comment]
I'd say leave it empty.

**Comment by azsam** [comment]
Thanks, I'll feedback if something goes wrong;-)

**Comment by tomfulton** [comment]
Yeah, I noticed this in Kenn's update too.  The ID gets generated automatically through the UI for new fieldsets.

I did run into an issue when creating some data programatically because I did not populate this ID in my JSON (caused issues in the UI adding/editing data).  I'm not sure if it was because I was using upload fields or if this needs to be populated for other situations as well.  I just fixed it by adding the `id` field to my data import via `Guid.NewGuid()`.  It's probably best to just populate this if creating data programatically, or maybe Archetype can be updated to add it for you in other cases when it's not there.

In general I think it's awesome and super-useful that each fieldset has a Unique ID now and solves some weird stuff I've had to do in the past :smile:  #h5yr
